### ISSUE TITLE
Authentication always fails

### ISSUE THREAD

**Issue by leehotae** [playgameservices/play-games-plugin-for-unity#1805]
I follow guide.
My app shows green dialog  but Authentication may be faild though I registered play console tester.
![capture](https://user-images.githubusercontent.com/21328168/27690476-61908012-5d1c-11e7-968f-716a52573cae.PNG)

first init code
`PlayGamesClientConfiguration config = new PlayGamesClientConfiguration.Builder().Build();
        PlayGamesPlatform.InitializeInstance(config);
        GooglePlayGames.PlayGamesPlatform.DebugLogEnabled = true;
        PlayGamesPlatform.Activate();`

and auth code

`if (!Social.localUser.authenticated)
        {
            Social.localUser.Authenticate((bool success) =>
            {
                if(success)
                    MainSystem.Instance.sceneManager.SceneLoad("test");
            });
        }`

what is the problem

sorry for my bad english

thank you for reading

**Comment by wixmax** [comment]
I'm having the same problem here. This problem is due to new updates on Google Play Console.

**Comment by leehotae** [comment]
@wismax

thanks for reply

Do you find any solution?

**Comment by interpol-kun** [comment]
Same problem.

**Comment by wixmax** [comment]
I found the solution   [here](https://github.com/playgameservices/play-games-plugin-for-unity/issues/1800#issuecomment-310877125)
It works successfully

**Comment by leehotae** [comment]
@wixmax 

thank you 
but I think it did not work for me

In my case 
default setting capture
![capture1](https://user-images.githubusercontent.com/21328168/27761557-316b09ec-5e99-11e7-8963-fd9417f00933.PNG)
![capture2](https://user-images.githubusercontent.com/21328168/27761558-33f635c4-5e99-11e7-90dc-aaf9a98cb4ee.PNG)

In your link
He says default sha1 in credential is upload certificate sha1 (2D:E3:15:F4:F2:5A...) and replace with  app certificate sha1 (62:4E:D0:B7:8E...)

but in my case 
default sha1 in credential is equal to app certificate sha1 

Is it right for my understanding your link article?

**Comment by interpol-kun** [comment]
It didn't help me too.
App is still trying to log in, but nothing happens.

**Comment by leehotae** [comment]
@interpol-kun 

Is your default sha1 in credential equal to app certificate sha1?

**Comment by interpol-kun** [comment]
@leehotae 

Yes, it is. I have attempting-login-window like you showed on the 1st screenshot, but there is no top pop-up gravity of successfull login, and Social.ShowAchievementsUI() doesn't work as well.

**Comment by interpol-kun** [comment]
@leehotae 

Listen carefully. Copy SHA-1 from "Upload certificate", not "App signing certificate".

Now it works fine!

**Comment by wixmax** [comment]
@leehotae 
Yes, as he said @interpol-kun 
Replace SHA-1 in https://console.developers.google.com/ with this

![933](https://user-images.githubusercontent.com/16684086/27762880-5d611a6c-5e69-11e7-972d-21fe35e5afa5.png)

**Comment by leehotae** [comment]
@interpol-kun 
thank you
it works fine too

**Comment by leehotae** [comment]
@wixmax 
thank you
it works fine too

**Comment by haydin20** [comment]
Create your APK with .Keystore. If you dont use .keystore login not success.

**Comment by PowercoderJr** [comment]
@haydin20 
Can you please explain to me what am I doing wrong? I created new .keystore, built .apk, loaded it to Google Play Console and tried both "Upload certificate" and "App signing certificate" SHA-1 prints. Still nothing works. :(

**Comment by PowercoderJr** [comment]
What am I doing wrong? I created new .keystore, built .apk, loaded it to Google Play Console and tried both "Upload certificate" and "App signing certificate" SHA-1 prints. E-mail is in testers list. Still nothing works. :(

**Comment by eyho** [comment]
Problem is, you copied your application's SHA-1 key to your API credentials. You need the upload certification SHA-1 key instead.

**Comment by GuillaumZ** [comment]
Not working here. Same problem. 
My SHA-1 is right. 
It's the good "upload certification SHA-1" from the google play console, that I can find, already well written, on the API console with the good App ID and package name.

When I build directly from Unity to my phone, the login to Google Play Games works.
But when I download the game from the Closed Beta Play Store, the login always fails.

I really don't know where to look after now to correct this.

**Comment by haydin20** [comment]
i solved the problem 
follow this 
[https://github.com/playgameservices/play-games-plugin-for-unity/issues/1842](url)

**Comment by GuillaumZ** [comment]
Yes, thank you, I'm putting here too my screenshot because the screenshot in Correan up that we can see 
above is WRONG (for me) 

![keystoresolution](https://user-images.githubusercontent.com/8102439/28427494-aced999a-6d76-11e7-9bca-080fc65d3a4f.jpg)

**Comment by ki4chan** [comment]
I am having the same problem. When I linked my app It had App signing SHA1 in API manager by default. Then the app was trying to sign in but failing every time. I replaced Upload Certificate with App Signing Certificate, now even signing in screen doesn't pop up.

One more question, are you guys using API key feature from API manager(the option above OAuth 2.0 client ID)?

**Comment by MTaimurAzhar** [comment]
@interpol-kun said Listen carefully. Copy SHA-1 from "Upload certificate", not "App signing certificate".
Thanks buddy, it helped me and I got it fixed!

**Comment by rico345100** [comment]
Doesn't work to me. I replaced both "App signing certificate" and "Upload certificate", still always fails.

**Comment by MTaimurAzhar** [comment]
Actually putting upload certificate into credentials works for an uploaded version.

**Comment by abramovks** [comment]
This helped me.

https://github.com/playgameservices/play-games-plugin-for-unity/issues/2217#issuecomment-417619382

Can confirm this works:
open template-AndroidManifest.txt
"\ APP_ID" to "\u003__APP_ID__"
setup GPGS from Unity (press Window/Google Play Services/Setup)



True, after the correction, I received the error "Authentification failed". But it was a problem with SHA-1. It was decided when I did
keytool -exportcert -keystore user.keystore -list -v
And put the correct SHA-1 in https://console.developers.google.com/apis/credentials in the application settings.

**Comment by kk3hi3123** [comment]
Does it mean that in:
Google Play Console > Game services > My application > Linked apps,
deleting the Android app authorized with the old OAuth2 Client ID (App signing certificate),
and then adding a new Android app authorized with the new OAuth2 Client ID (Upload certificate)?

I still cannot fix it after that. Anyone helps?

**Comment by mohammadmahdi123** [comment]
I tested both Upload certificate and app signing certificate and with both of them, I can not sign in.
My app sha1 (keystore) is the same as upload certificate. I changed sha1 in console.developers.google.com to Upload certificate sha1 and app signing certificate sha1.

**Comment by tekminewe** [comment]
I tried "Upload Certificate" previously and able to authenticate, but it's failed after few install. My final solution is to have both "Upload Certificate" and "App Signing Certificate" sha-1 in the google console and it's working fine now

**Comment by SniperED007** [comment]
Under your Linked Apps section in Google Play Console -> Game Services you need to add 2 apps, 1 for production (the one that gets released via Google Play) and use the "App Signing" SHA-1,
and the second app is for testing (where you build directly to your phone) and for that one use the "Upload Certificate" SHA-1.

**Comment by rafaelf82** [comment]
> Under your Linked Apps section in Google Play Console -> Game Services you need to add 2 apps, 1 for production (the one that gets released via Google Play) and use the "App Signing" SHA-1,
> and the second app is for testing (where you build directly to your phone) and for that one use the "Upload Certificate" SHA-1.

This should be mentioned on the setup instructions. Worked for me. Not sure yet if it had any side effects.

**Comment by arsalankhanji** [comment]
Social.localUser.Authenticate was working fine for me.

However, since the last couple of days, it has suddenly stopped working (authentication failing). I didn't change anything at all. Is someone else also experiencing the same issue?

**Comment by smokinpuppy** [comment]
Same Problem here

**Comment by dzad** [comment]
hello I was having this issue last 3 days and I found the solution.
so if you are downloading the app from google play use the signin sha1. if not use the uploading sha1. 
and the thing that you shouldn't ignore is to update the oauth consent screen
![](https://i.imgur.com/4EHwOP8.jpg)
![](https://i.imgur.com/IlmZ3C6.jpg)

update one of those fileds and hit save and boom the google play services works :wink:

**Comment by NullOne01** [comment]
I had simillar problem, when I was using Saved Games. My problem was that I didn't tell Google Services that I want to use Saved Games. Solution is to find this options in Google Services in Google Play Developer Console.

**Comment by manwithsteelnerves** [comment]
Taken from [Cross Platform Native Plugins](http://u3d.as/1szE)
But, the info here is very general and works with any plugins.

## Why sign in fails? 

For Google Play Services to sign in successfully, you need to make sure the apk from where you are logging in should be allowed to make requests to google servers.

## How to make the APK authorized to make requests to google servers?

Each apk built with a keystore has a SHA fingerprint. You need to make sure the SHA fingerprint is added in the google cloud.

You can fetch it with keytool command (check [here](https://assetstore.essentialkit.voxelbusters.com/game-services/faq#what-are-different-sha-fingerprints-that-needs-to-be-used-for-logging-in-successfully-on-different-environments-and-how-to-create-them) on how to get SHA for each environment )

## How many SHA fingerprints I need to add for successful authentication?
One per each environment. In total 3 environments (dev, release and production)
1. Development build uses android default debug keystore.

2. Release build uses the keystore you set in publishing settings of player settings

3. Google play store build which uses google play signing.

If you want all the above builds to authenticate successfully, add all 3 fingerprints. Else only the required environment.

## How to get each environment's SHA?

1. Debug ( keytool -list -v -keystore "PATH_TO_DEBUG_KEYSTORE" -alias androiddebugkey -storepass android -keypass android)

2. Release (Get from Google play console -> Your App -> Setup -> App signing -> Upload key certificate )

3. Apk/AAb downloaded from Google play store (Get from Google play console -> Your App -> Setup -> App signing -> App signing key certificate )

## Where to add these SHA fingerprints?
Google Play Console. Check [here](https://assetstore.essentialkit.voxelbusters.com/game-services/setup/android#adding-a-sha-fingerprint) for details on how to add a SHA fingerprint.

**Comment by berkun5** [comment]
So after 7 hours of research if your problem is not SHA fingerprints or any other copy-paste stuff then it might be this: 
![Screenshot_22](https://user-images.githubusercontent.com/80388989/119701685-53143000-be5d-11eb-8218-c193713f3607.png)

So you see on 2021 version of google play console select your app then on the left menu list, navigate to "**Play Games Services**" > "**Setup and management**" > **Configuration**. 

You will see your "Credentials" (if you haven't created them already, you need them first).

Your credential status has to be "Published". This is something **different than publishing the app/game as "Closed/Internal" testing**. If you are seeing the status as "draft" then it will not work.

You need to **publish those credentials** in order to actually log in.

**Comment by ariok** [comment]
> So after 7 hours of research if your problem is not SHA fingerprints or any other copy-paste stuff then it might be this:
> ![Screenshot_22](https://user-images.githubusercontent.com/80388989/119701685-53143000-be5d-11eb-8218-c193713f3607.png)
> 
> So you see on 2021 version of google play console select your app then on the left menu list, navigate to "**Play Games Services**" > "**Setup and management**" > **Configuration**.
> 
> You will see your "Credentials" (if you haven't created them already, you need them first).
> 
> Your credential status has to be "Published". This is something **different than publishing the app/game as "Closed/Internal" testing**. If you are seeing the status as "draft" then it will not work.
> 
> You need to **publish those credentials** in order to actually log in.

This is completely wrong. 
From the official documentation I read that draft credentials can be used without any problem. 

From the Doc: "After you click Done in the dialog, the available Client IDs will refresh. Choose the credential you created from the drop-down menu and then click Save Changes. This creates the credential as a draft, enabling you to authenticate to Play Games Services in your game." 

https://developers.google.com/games/services/console/enabling

**Comment by olehkuznetsov** [comment]
You can sign-in with Draft configuration if you have added your account in testers list.
OAuth screen in cloud console has its own testers list and need to be published or you need to add same account to OAuth screen testers list.
